File: pixo-c#/DalSql/models/Product.cs
using System;
using System.Collections.Generic;

namespace DalSql.models;

public partial class Product
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public decimal? Price { get; set; }

    public string? ImageUrl { get; set; }

    public int? CategoryId { get; set; }

    public int? SizeId { get; set; }

    public int? TypeId { get; set; }

    public virtual Category? Category { get; set; }

    public virtual ICollection<Customization> Customizations { get; set; } = new List<Customization>();

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();

    public virtual Size? Size { get; set; }

    public virtual Type? Type { get; set; }
}

File: projectAngular/src/app/models/product.ts
export class Product {

    
  id: number;                // מזהה המוצר
  name: string;              // שם המוצר
  price: number;             // מחיר המוצר
  imageUrl: string;          // כתובת ה-URL של התמונה
  categoryId: number;        // מזהה הקטגוריה
  sizeId: number;     
  typeId: number;     //תת קטגוריה

  constructor(id: number,name: string,price: number,imageUrl: string,categoryId: number,sizeId: number,typeId:number
) {
    this.id = id;
    this.name = name;
    this.price = price;
    this.imageUrl = imageUrl;
    this.categoryId = categoryId;
    this.sizeId = sizeId;
    this.typeId = typeId;
  }

}


